### PR TITLE
Make `getIdsFromQuery` consider `0` as a valid id.

### LINKED ISSUE #6980: `getIdsFromQuery` does not get `0` as Id.
**Describe the bug**
I'm trying to use `@woocommerce/navigation.getIdsFromQuery` to parse Ids from a query param. The param can take values like:
- `param=0`
- `param=0,5677`
In both cases `0` is ignored.

**To Reproduce**

```js
getIdsFromQuery( '0' );
getIdsFromQuery( '12,0,67' );
```

**Expected behavior**
`0` as a valid `Number` should be a valid Id.
Therefore, I should get `[0]` and `12,0,67` respectively.

### Technical

I guess at https://github.com/woocommerce/woocommerce-admin/blob/main/packages/navigation/src/index.js#L97 we should use `( id ) => ! isNaN( id )` instead of `Boolean`, as `parseInt` already identified numbers and not-numbers.

### PR DESCRIPTION
**Pull request by tomalec**
- Make `getIdsFromQuery` consider `0` as a valid id. (bcfab01)
	Previously `Boolean` filter was removing it from the list.
	Fixes #6980
- Add basic set of tests for getIdsFromQuery, as they were none. (aa17f4e)
- Replace `lodash.uniq` with ES6 equivalent, (7eb2eb4)
- Update JSDoc for getIdsFromQuery, to specify default value, returned type and description(b817f71)

### Accessibility

n/a

### Screenshots
n/a

### Detailed test instructions:

```js
getIdsFromQuery( '0' );
getIdsFromQuery( '12,0,67' );
```

Or go to any page that fetches Ids from URL, and change to URL to contain
```
param=0
param=0,5677
```
see how it's parsed.

### Doubts:

I wasn't sure if `parseInt` is the best choice, and if the following cases are desired:
```
getIdsFromQuery('8.5,8.6');    // [8]
getIdsFromQuery('0xF');    // [0]
getIdsFromQuery("32px");   // [32]
getIdsFromQuery("5e1");    // [5]
getIdsFromQuery('12 345'); // [12]
```
Left this as is, not to introduce too many breaking changes.

----

no changelog necessary as it is in navigation package

### CODE REVIEW

**Comment by tomalec** [comment]
I was also wondering about changing the API, or providing new util that returns `Set`, without transforming it to an array.

In many cases when we use it like:
- `getIdsFromQuery( foo ).includes( id )`
- `new Set( getIdsFromQuery( foo ) )`
which effectively is `new Set( [ ...new Set( array ) ] )`

If we make it return just set, we make:
- simple operations simple and performant
- advanced operations, not much complicated: `[...getIdsFromQuery(foo)].forEach`

------


We could also expose both, for backward compatibility:
- `getIdsSetFromQuery: Set<string>`
- `getIdsFromQuery: Array<string>` like:  `( qs ) => [...getIdsSetFromQuery(qs)]`

File: readme.txt [hunk 1]
```diff
@@ -74,6 +74,8 @@ Release and roadmap notes are available on the [WooCommerce Developers Blog](htt
 == Changelog ==
 
 == Unreleased ==
+
+- Fix: `@woocommerce/navigation.getIdsFromQuery` support for `0` as a valid id. #7028
```
Comment (adrianduffell) [inline readme.txt:78]: We don't need a changelog item for WC Admin as it doesn't directly impact it- i.e. it's not fixing a bug in WC Admin.

File: packages/navigation/src/index.js [hunk 2]
```diff
@@ -86,16 +86,18 @@ export const getScreenFromPath = ( path = getPath() ) => {
 /**
  * Get an array of IDs from a comma-separated query parameter.
  *
- * @param {string} queryString string value extracted from URL.
- * @return {Array} List of IDs converted to numbers.
+ * @param {string} [queryString=''] string value extracted from URL.
+ * @return {Array<number>} List of IDs converted to an arry of unique integers.
```
Comment (adrianduffell) [inline packages/navigation/src/index.js:90]:
```suggestion
 * @return {Array<number>} List of IDs converted to an array of unique integers.
```

Also should the return type still be an array when it uses `Set`?

**Review by adrianduffell — DISMISSED** [review]
@tomalec Thanks for contributing the PR, it looks good! 

> I was also wondering about changing the API, or providing new util that returns `Set`, without transforming it to an array.

Ideally we would keep backwards compatibility with the existing API. I'm not very familiar with `Set`. How compatible would it be with existing code expecting an array being returned? 

If it has compatibility issues, I like your idea about providing a new util.

I made a few minor suggestions in the code.

Thanks for writing the tests!

File: packages/navigation/src/index.js (hunk 2 above)
Comment (tomalec) [reply packages/navigation/src/index.js:90]: It uses `Set` internally but still returns an array https://github.com/woocommerce/woocommerce-admin/pull/7028/files/690e5d73bd611674831e11f69f29e1a832c52341#diff-6efa564f97a1e5bbd69c59abf45e53f218b9984f77817a13cbdfd4b64de57518R93

**Comment by tomalec** [comment]
> Ideally we would keep backwards compatibility with the existing API. I'm not very familiar with Set. How compatible would it be with existing code expecting an array being returned?

[`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) is a native JS object, that lets you store unique values in a performant and convenient way.

It's [iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/@@iterator), but not compatible with `Array` and does not implement the majority of its API, but could be easily cast to one:
- `Array.from( mySet )`
- `[ ...mySet ]`

**Comment by tomalec** [comment]
I made another branch with `getIdsSetFromQuery` proposal at https://github.com/woocommerce/woocommerce-admin/compare/fix/6980-getIdFromQuery...fix/6980-getIdFromQuery-set?expand=1

**Comment by adrianduffell** [comment]
> Just to make it clear, the changes proposed [here](https://github.com/woocommerce/woocommerce-admin/pull/7028#issue-648649746) do not change the API of `getIdsFromQuery` and are fully backward-compatible for the environments that supports `new Set( [1,2] )`.

Oh, sorry I didn't notice that you cast the `Set` to an `Array` originally. Thanks for clarifying, I get it now 😅 

> I made another branch with getIdsSetFromQuery proposal at https://github.com/woocommerce/woocommerce-admin/compare/fix/6980-getIdFromQuery...fix/6980-getIdFromQuery-set?expand=1

This approach LGTM. I would maybe just change the name of the function to `getIdsFromQueryAsSet`.

File: packages/navigation/README.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ Gets time related parameters from a query.
 ### getIdsFromQuery(queryString) ⇒ <code>Array</code>
 Get an array of IDs from a comma-separated query parameter.
 
-**Returns**: <code>Array</code> - List of IDs converted to numbers.
+**Returns**: <code>Array</code> - List of IDs converted to an arry of unique integers.
```
Comment (samueljseay) [inline packages/navigation/README.md:48]: *an array

**Comment by tomalec** [comment]
I fixed the typo. Is this PR good to go now? 

I think the new `getIdsSetFromQuery`/`getIdsFromQueryAsSet`, could be handled in a separate PR [`fix/6980-getIdFromQuery-set`](https://github.com/woocommerce/woocommerce-admin/compare/fix/6980-getIdFromQuery...fix/6980-getIdFromQuery-set?expand=1)

**Review by samueljseay — APPROVED** [review]
This looks great, thanks @tomalec 🚀
